sbi/utils/metrics.py

```diff
@@ -262,6 +262,84 @@ def unbiased_mmd_squared_hypothesis_test(x, y, alpha=0.05):
     return mmd_square_unbiased, threshold
 
 
+def posterior_shrinkage(prior_samples, post_samples):
+    """
+    Calculate the posterior shrinkage, quantifying how much
+    the posterior distribution contracts from the initial
+    prior distribution.
+
+    Parameters
+    ----------
+    prior_samples : array-like or torch.Tensor
+        Samples from the prior distribution.
+    post_samples : array-like or torch.Tensor
+        Samples from the posterior distribution.
+
+    Returns
+    -------
+    shrinkage : float or array-like
+        The posterior shrinkage.
+    """
+
+    if len(prior_samples) == 0 or len(post_samples) == 0:
+        raise ValueError("Input samples are empty")
+
+    if isinstance(prior_samples, torch.Tensor):
+        prior_samples = prior_samples.cpu().numpy()
```

Comment:
Let's keep it as torch tensor, if we do not need any special numpy functionality